impl/src/main/java/com/okta/sdk/impl/oauth2/AccessTokenRetrieverServiceImpl.java

```diff
@@ -141,6 +141,7 @@ String createSignedJWT() throws InvalidKeyException, IOException {
             .setIssuer(clientId)
             .setSubject(clientId)
             .claim("jti", UUID.randomUUID().toString())
+            .setHeaderParam("kid", clientId + privateKey.hashCode())
```

Comment:
This should match the `kid` in the jwks endpoint, or the one used when the key was uploaded:
https://developer.okta.com/docs/guides/implement-oauth-for-okta-serviceapp/create-serviceapp-grantscopes/

The `kid` here seem arbitrary?

Please let me know if I'm not understanding something correctly

Reply:
@sergiishamrai-okta The best way to go about this would be to add a `kid` parameter to the [YAML configuration](https://github.com/okta/okta-sdk-java#yaml-configuration) under `okta.client` (like `okta.client.kid`). Then, we could simply set the same `kid` value here while constructing the JWT. 

Also, refer related issue: https://github.com/okta/oss-technical-designs/pull/29#issuecomment-767079972